tests/generic_config_updater/test_syslog.py

```diff
@@ -0,0 +1,267 @@
+import logging
+import pytest
+
+from tests.common.helpers.assertions import pytest_assert
+from tests.common.config_reload import config_reload
+from tests.generic_config_updater.gu_utils import apply_patch, expect_op_success_and_reset_check, expect_res_success, expect_op_failure
+from tests.generic_config_updater.gu_utils import generate_tmpfile, delete_tmpfile
+
+pytestmark = [
+    pytest.mark.topology('t0'),
+]
+
+logger = logging.getLogger(__name__)
+
+# This is restricted by sonic-syslog.yang
+SYSLOG_MAX_SERVER=-1
+
+SYSLOG_THRESHOLD=10
+SYSLOG_INTERVAL=1
+
+@pytest.fixture(scope="module")
+def cfg_facts(duthosts, rand_one_dut_hostname):
+    duthost = duthosts[rand_one_dut_hostname]
+    return duthost.config_facts(host=duthost.hostname, source="persistent")['ansible_facts']
+
+@pytest.fixture(scope="module")
+def setup_env(duthosts, rand_one_dut_hostname, cfg_facts):
+    """
+    Setup/teardown fixture for syslog config
+    Args:
+        duthosts: list of DUTs.
+        rand_selected_dut: The fixture returns a randomly selected DuT.
+        cfg_facts: config facts for selected DUT
+    """
+    duthost = duthosts[rand_one_dut_hostname]
+
+    config_tmpfile = generate_tmpfile(duthost)
+    logger.info("config_tmpfile {}".format(config_tmpfile))
+    logger.info("Backing up config_db.json")
+    duthost.shell("sudo cp /etc/sonic/config_db.json {}".format(config_tmpfile))
+
+    # Cleanup syslog server config
+    syslog_servers = cfg_facts.get('SYSLOG_SERVER', {})
+    for syslog_server in syslog_servers:
+        duthost.shell("sudo config syslog del {}".format(syslog_server),
```

Comment:
>shell [](http://example.com/codeflow?start=16&length=5)

Check the return code?

Reply:
Added the check. But I am confused that when do we need to check the return code? Seems not every command is being checked.

Reply:
Ideally we need to check every return code in case of any disaster. In some special case, if you don't care about any disaster/corner cases and just want to continue, then it's okay to ignore the return code.

For example, in this case, `config` command will possible return an error code if the system is so busy. You should treat it as a test case failure.